### ISSUE TITLE
Generic conformal: data leakage and multiclass one-hot probabilities

### ISSUE THREAD

**Issue by sipemu** [sipemu/fdars#20]
## Summary

Two issues in the generic conformal prediction module (`src/conformal/generic.rs`):

### 1. Data leakage in generic conformal regression/classification

`conformal_generic_regression` and `conformal_generic_classification` take a pre-fitted model and compute calibration residuals on a random subset of the training data. However, the model was trained on ALL data including this calibration subset, so:

- Calibration residuals are **in-sample** and systematically too small
- Prediction intervals are too narrow (optimistic)
- The distribution-free coverage guarantee `P(y ∈ C) ≥ 1 − α` is broken

The Rust doc comments already acknowledge this (lines 17–23 of `generic.rs`), but users of the public API may not realize it.

**Possible fixes:**
- **Option A (recommended):** Accept a `proper_training_indices` or `calibration_indices` parameter so the user can supply a held-out calibration set that was NOT used for training. The model is fitted on the proper training set; calibration is done on held-out data.
- **Option B:** Add `refit: bool` parameter — when true, refit the model on the proper training subset. This requires `FpcPredictor` to support fitting (new trait method).
- **Option C (minimal):** Return a warning flag in the result struct (e.g., `data_leakage_warning: bool = true`) so downstream code can surface it.

### 2. Multiclass one-hot probabilities in generic classification

For multiclass (`n_classes > 2`), `predict_from_scores` returns a class label as `f64` (e.g., `2.0`). The code converts this to a one-hot probability vector at `generic.rs:162–169`:

```rust
let c = pred.round() as usize;
let mut probs = vec![0.0; n_classes];
if c < n_classes {
    probs[c] = 1.0;
}
```

With one-hot probabilities:
- **LAC scores** are either `0.0` (correct prediction) or `1.0` (wrong), with no gradation
- **Prediction sets** are degenerate: either just the predicted class (size 1) or all classes (size K)
- The conformal sets provide no meaningful uncertainty information

**Possible fixes:**
- **Option A (recommended):** Add a `predict_probabilities(&self, scores: &[f64], scalars: Option<&[f64]>) -> Vec<f64>` method to the `FpcPredictor` trait that returns a full probability vector for all classes. For `FunctionalLogisticResult` (binary), this would return `[1-p, p]`. For multiclass models, proper class probabilities.
- **Option B:** Restrict `conformal_generic_classification` to binary classification only and return an error for multiclass.

### Affected versions

Introduced in the conformal module (v0.8.x).

### Impact

- Binary classification via the generic path works correctly (the `n_classes == 2` branch uses `[1-pred, pred]` which gives proper probabilities)
- Split/CV+ classification paths using LDA/QDA/kNN are unaffected (they compute proper probability vectors directly)
- Only the generic conformal path is affected
